src/AttributeEngine/AttributeEngineServiceProvider.php

```diff
@@ -31,6 +31,8 @@ public function register()
             if (is_array(config('opendialog.attribute_engine.custom_attributes'))) {
                 $attributeResolver->registerAttributes(config('opendialog.attribute_engine.custom_attributes'));
             }
+            // Gets custom dynamic attributes from the database
+            $attributeResolver->registerAllDynamicAttributes();
```

Comment:
I think we should keep this consistent with existing attribute registration (as seen with core and app attributes above), with something similar to:

```suggestion
            if ($this->hasDynamicAttributes()) {
                $attributeResolver->registerAttributes($this->formatDynamicAttributes());
            }
```

This way the resolver remains fairly agnostic to the source of registered attributes. The outcome of this change is to create some `formatDynamicAttributes` method that can retrieve dynamic attributes from the database and map them the same format as found in the config files:

```php
[
    'my_custom_str_attr' => StringAttribute::class,
    'my_custom_bool_attr' => BooleanAttribute::class,
    ...
]
```

For this we can make use of `AttributeTypeService.getAttributeTypeClass` to turn attribute type IDs (eg. `attribute.core.string`) into attribute type classes (eg. `StringAttribute::class`).

Reply:
This has been updated.